common/events/player_event_logs.cpp

```diff
@@ -123,21 +134,466 @@ void PlayerEventLogs::ProcessBatchQueue()
+				out.merchant_type           = in.merchant_type;
+				out.item_id                 = in.item_id;
+				out.item_name               = in.item_name;
+				out.charges                 = in.charges;
+				out.cost                    = in.cost;
+				out.alternate_currency_id   = in.alternate_currency_id;
+				out.player_money_balance    = in.player_money_balance;
+				out.player_currency_balance = in.player_currency_balance;
+				out.created_at              = r.created_at;
+
+				if (m_etl_settings.contains(PlayerEvent::EventType::MERCHANT_PURCHASE)) {
+					r.etl_table_id = m_etl_settings.at(PlayerEvent::EventType::MERCHANT_PURCHASE).next_id;
+					m_etl_settings.at(PlayerEvent::EventType::MERCHANT_PURCHASE).next_id++;
+				}
+
+				etl_queues.merchant_purchase.push_back(out);
+				break;
+			}
+			case PlayerEvent::EventType::NPC_HANDIN: {
+				PlayerEvent::HandinEvent in{};
+				PlayerEventNpcHandinRepository::PlayerEventNpcHandin out{};
+				PlayerEventNpcHandinEntriesRepository::PlayerEventNpcHandinEntries out_entries{};
+
+				{
+					std::stringstream ss;
+					ss << r.event_data;
+					cereal::JSONInputArchive ar(ss);
+					in.serialize(ar);
+				}
+
+				out.npc_id          = in.npc_id;
+				out.npc_name        = in.npc_name;
+				out.handin_copper   = in.handin_money.copper;
+				out.handin_silver   = in.handin_money.silver;
+				out.handin_gold     = in.handin_money.gold;
+				out.handin_platinum = in.handin_money.platinum;
+				out.return_copper   = in.return_money.copper;
+				out.return_silver   = in.return_money.silver;
+				out.return_gold     = in.return_money.gold;
+				out.return_platinum = in.return_money.platinum;
+				out.is_quest_handin = in.is_quest_handin;
+				out.created_at      = r.created_at;
+
+				if (m_etl_settings.contains(PlayerEvent::EventType::NPC_HANDIN)) {
+					r.etl_table_id = m_etl_settings.at(PlayerEvent::EventType::NPC_HANDIN).next_id;
+					m_etl_settings.at(PlayerEvent::EventType::NPC_HANDIN).next_id++;
+				}
+
+				if (!in.handin_items.empty()) {
+					for (auto const &i: in.handin_items) {
+						out_entries.charges                    = i.charges;
+						out_entries.evolve_amount              = 0;
+						out_entries.evolve_level               = 0;
+						out_entries.item_id                    = i.item_id;
+						out_entries.player_event_npc_handin_id = r.etl_table_id;
+						out_entries.type                       = 1;
+						out.created_at                         = r.created_at;
+
+						if (!i.augment_ids.empty()) {
+							uint32 augments[6]{};
+							for (int x = 0; x < i.augment_ids.size(); x++) {
+								augments[x] = i.augment_ids[x];
+							}
+							out_entries.augment_1_id = augments[0];
+							out_entries.augment_2_id = augments[1];
+							out_entries.augment_3_id = augments[2];
+							out_entries.augment_4_id = augments[3];
+							out_entries.augment_5_id = augments[4];
+							out_entries.augment_6_id = augments[5];
+						}
+					etl_queues.npc_handin_entries.push_back(out_entries);
```

Comment:
While we know the 'entries' sizes, there could be other events of the same type in the batch queue, so we would need to cycle through the batch_queue to parse the total size for the potential multiple events.  Would this be beneficial, instead of the dynamic size adjustments as the vectors are add to?  If so, happy to do that.
Same comment for all the reserve items.

Reply:
Yes it would be beneficial as it will make a difference with servers with high event volume.

Reply:
Added reserve to all etl_queues.

Reply:
Thank you sir